### ISSUE TITLE
better from derive for new types

### ISSUE THREAD

**Issue by Luro02** [JelteF/derive_more#90]
generated code:
``` rust
#[derive(From)]
pub type SignedDecimalFloatingPoint(f64);

impl ::std::convert::From<(f64)> for SignedDecimalFloatingPoint {
    fn from(original: (f64)) -> SignedDecimalFloatingPoint {
        SignedDecimalFloatingPoint(original)
    }
}
```

better:
``` rust
impl<T> From<T> for SignedDecimalFloatingPoint
where
    T: Into<f64>,
{
    fn from(value: T) -> Self {
        Self(value.into())
    }
}
```
This will allow every type, that implements `Into<f64>`.

**Comment by JelteF** [comment]
I've tried doing this, but it doesn't compile if the newtype has a generic type.

**Comment by JelteF** [comment]
Support for forwarding the from call is now added on master:
```rust
#[derive(From)]
#[from(forward)]
pub type SignedDecimalFloatingPoint(f64);
```

You can even do this to only forward the from call for one of the fields:
```rust
#[derive(From)]
pub type TwoDecimalFloatingPoint{
    #[from(forward)]
    a: f64,
    #[from]
    b: f64,
);
```

**Comment by JelteF** [comment]
This is now released as version 0.99.0. Announcement link: https://reddit.com/r/rust/comments/duptc0/derive_more_0990_released_biggest_release_ever
